pkg/machines/scripts/create_machine.sh

```diff
@@ -1,12 +1,14 @@
 #!/bin/sh
 set -eu -o noglob
 
-VM_NAME="$1"
-SOURCE="$2"
-OS="$3"
-MEMORY_SIZE="$4" # in MiB
-STORAGE_SIZE="$5" # in GiB
-START_VM="$6"
+CONNECTION_URI=""
+[[ !  -z  $1 ]] && CONNECTION_URI="qemu:///$1" # ex: qemu:///system
```

Comment:
Unless you call this script completely without arguments, this should *never* happen. And if you do, it will bail out because `$2` will also not exist. Also, you would need to quote the `$1` in the test. Finally, `[[` is a bashism, please use regular POSIX `[ ]` to also work  in Debian.

But I don't think this makes sense to check at all. If you would forget to specify the first argument, then the VM name would silently be used as connection URI, and stuff would still break in a readable way.

Reply:
Fixed.